pkg/kbld/image/preresolved.go

```diff
@@ -3,21 +3,34 @@
 
 package image
 
+import (
+	"github.com/k14s/kbld/pkg/kbld/config"
+)
+
 type PreresolvedImage struct {
-	url string
+	url   string
+	metas []config.ImageMeta
 }
 
 type PreresolvedImageSourceURL struct {
 	Type string // always set to 'preresolved'
 	URL  string
+	Tag  string `json:",omitempty" yaml:",omitempty"`
 }
 
 func (PreresolvedImageSourceURL) meta() {}
 
-func NewPreresolvedImage(url string) PreresolvedImage {
-	return PreresolvedImage{url}
+func NewPreresolvedImage(url string, metas []config.ImageMeta) PreresolvedImage {
+	return PreresolvedImage{url, metas}
 }
 
 func (i PreresolvedImage) URL() (string, []Meta, error) {
+	if len(i.metas) > 0 {
+		var givenMetas []Meta
+		for _, m := range i.metas {
+			givenMetas = append(givenMetas, PreresolvedImageSourceURL{Type: "preresolved", URL: m.URL, Tag: m.Tag})
+		}
+		return i.url, givenMetas, nil
+	}
```

Comment:
Rename givenMetas -> imageMetas? Since 'given' doesn't map to anything related to this domain.